src/notebooksyncagent.cpp

```diff
@@ -410,604 +470,501 @@ void NotebookSyncAgent::sendLocalChanges()
-            it = localInserted->erase(it);
         } else {
-            LOG_DEBUG("Entirely new addition:" << incidence->uid() << incidence->recurrenceId().toString());
-            ++it;
+            // Set the URI and the ETAG property to the required values.
+            LOG_DEBUG("Adding URI and ETAG to added incidence:" << incidence->uid() << incidence->recurrenceId().toString());
+            setIncidenceHrefUri(incidence, href);
+            incidence->setCustomProperty("buteo", "etag", m_updatedETags[href]);
+            Reader::CalendarResource resource;
+            resource.href = href;
+            resource.incidences = KCalCore::Incidence::List() << incidence;
+            KCalCore::ICalFormat icalFormat;
+            resource.iCalData = icalFormat.toICalString(IncidenceHandler::incidenceToExport(incidence));
+            m_receivedCalendarResources.append(resource);
         }
     }
 
-    for (KCalCore::Incidence::List::iterator it = localModified->begin(); it != localModified->end();) {
-        KCalCore::Incidence::Ptr sourceIncidence = *it;
-        const KCalId &kcalid = KCalId(sourceIncidence);
-        if (remoteDeletedIncidences.contains(kcalid) || mRemoteModifiedIds.contains(kcalid)) {
-            LOG_DEBUG("Discarding modification,"
-                      << (remoteDeletedIncidences.contains(kcalid) ? "was already deleted on server" : "")
-                      << (mRemoteModifiedIds.contains(kcalid) ? "was already modified on server": ""));
-            it = localModified->erase(it);
-            continue;
-        } else if (modifications.contains(kcalid)) {
-            KCalCore::ICalFormat iCalFormat;
-            KCalCore::Incidence::Ptr receivedIncidence = iCalFormat.fromString(modifications[kcalid]);
-            if (receivedIncidence.isNull()) {
-                LOG_WARNING("Not sending modification, cannot parse the received incidence:" << modifications[kcalid]);
-                it = localModified->erase(it);
-                continue;
-            }
-            // If incidences are the same, then we assume the local incidence was not changed after
-            // the remote incidence was received, and thus there are no modifications to report.
-            IncidenceHandler::prepareImportedIncidence(receivedIncidence);  // ensure fields are updated as per imported incidences
-            if (IncidenceHandler::copiedPropertiesAreEqual(sourceIncidence, receivedIncidence)) {
-                LOG_DEBUG("Discarding modification" << kcalid.toString());
-                it = localModified->erase(it);
-                continue;
-            }
-        }
-
-        // The default storage implementation applies the organizer as an attendee by default. Don't do this
-        // as it turns the incidence into a scheduled event requiring acceptance/rejection/etc.
-        const KCalCore::Person::Ptr organizer = sourceIncidence->organizer();
```

Comment:
Yep, looks like removePossibleLocalModificationIfIdentical() should be checking against the incidenceToExport() version. How is it working without doing that actually? I would have thought the dates wouldn't match. Yep, if the exported version is checked then the "remove organizer from attendees" logic can be moved to incidenceToExport(). Also that will make the copiedPropertiesAreEqual() function redundant.